locksmith/migrations/20191001184934-usermetadataConstraint.js

```diff
@@ -1,16 +1,9 @@
-'use strict';
+'use strict'
 
-const table ='UserTokenMetadata'
+const table = 'UserTokenMetadata'
 
 module.exports = {
-  up: async (queryInterface, Sequelize) => {
-    return queryInterface.addConstraint(table, ['tokenAddress', 'userAddress'],{
-      type: 'unique',
-      name : 'token_user_address_unique_constraint'
```

Comment:
these constraints break upsert on sequelize v6

Reply:
Please do it in another migration instead of modifying existing one because they won't be applied properly and re-check.

Reply:
@m00n620 I think @searchableguy is right. We should not modify older migrations but add new ones ...

Reply:
@m00n620

Sequelize moved away from using temporary replace function in v5 to `INSERT ON CONFLICT DO UPDATE` on postgres in v6. This requires having a unique constraint or a primary key.

We need the composite constraint because token address or user address on their own are not unique. A user address can have data on multiple locks so we require a compose constraint here. Anywhere we use upsert with usermetadata, we pass both token address as well as user address so that should work, right?

Could you please tell me where it breaks if you know or remember?

Reply:
@m00n620 Just to confirm, this was because upsert had a validation error due to the unique constraints. 

Did you check if it was an existing bug or issue? I found https://github.com/sequelize/sequelize/issues/13734

after debugging the raw queries. Those were all using primary key on ON_CONFLICT regardless if you passed a primary or not in upsert. So I'm passing conflict field manually now. 

Removing the constraints here would lead to inserting new data instead of updating old data since the id is not passed and we depend on userAddress and tokenAddress as the unique key for conflict in our codebase. 

Let me know if I'm correct.